server/rooms.ts

```diff
@@ -1018,7 +1018,12 @@ export class GlobalRoom extends BasicRoom {
 }
 
 export class BasicChatRoom extends BasicRoom {
-	log: Roomlog;
+	readonly log: Roomlog;
+	readonly autojoin: boolean;
+	readonly staffAutojoin: string | boolean;
+	readonly banwords: string[];
+	readonly uptime: number | null;
```

Comment:
I'm surprised this isn't modified. This should probably be renamed to like `creationTime` or something.

Reply:
Why are you surprised that it isn't modified? It's used in groupchats (for `/groupchatuptime`) and is just `Date.now()` (or `null` in the case that the room isn't a groupchat). I see no reason anyone should be mutating it.

Reply:
`now` isn't the uptime, `now - creationTime` is the uptime. The main thing that surprises me is that it's named `uptime` when it very clearly isn't the uptime but the creation time.